tools/pipelines/templates/build-npm-client-package.yml

```
@@ -230,6 +230,12 @@ extends:
                 lfs: '${{ parameters.checkoutSubmodules }}'
                 submodules: '${{ parameters.checkoutSubmodules }}'
 
+              - script: |
+                  echo "commit sha: $(git rev-parse HEAD)"
+                  echo "##vso[task.setvariable variable=COMMIT_SHA;isOutput=true]$(git rev-parse HEAD)"
```

Since ADO already provides the value of the commit that was checked out, I think it makes sense to use that directly.
```suggestion
                  echo "commit sha: $(Build.SourceVersion)"
                  echo "##vso[task.setvariable variable=COMMIT_SHA;isOutput=true]$(Build.SourceVersion)"
```